platforms/opencl/src/OpenCLContext.cpp

```diff
@@ -149,6 +150,9 @@ OpenCLContext::OpenCLContext(const System& system, int platformIndex, int device
                 if (devices[i].getInfo<CL_DEVICE_TYPE>() != CL_DEVICE_TYPE_GPU) {
                     processingElementsPerComputeUnit = 1;
                 }
+                else if (vendor.size() >= 5 && vendor.substr(0, 5) == "Apple") {
+                  processingElementsPerComputeUnit = 4 * 32;
+                }
```

Comment:
`vendor` is undefined here. `platformVendor` corresponds to `CL_PLATFORM_VENDOR` and `vendor` would be `CL_DEVICE_VENDOR` to match variable names after Line 216.

Reply:
I made a new variable called `deviceVendor` to explicitly differentiate the two versions of "vendor". This differs from the subsequent control flow block.